crates/providers/src/provider.rs

```diff
@@ -31,223 +33,432 @@ pub struct Provider<T: Transport = BoxTransport> {
     from: Option<Address>,
```

Comment:
We don't need this special case in a separate struct, and instead implement the provider trait on rpcclient